src/schedule/zephyr_ll.c

```diff
@@ -122,30 +122,18 @@ static void zephyr_ll_task_insert_after_unlocked(struct task *task, struct task
 
 static inline enum task_state do_task_run(struct task *task)
 {
-	uint32_t cycles0, cycles1, diff;
 	enum task_state state;
 
-	cycles0 = k_cycle_get_32();
+#if CONFIG_PERFORMANCE_COUNTERS
+	perf_cnt_init(&task->pcd);
+#endif
```

Comment:
@lgirdwood We agreed with @btian1 to get rid of the "always-built-in" traces task level state. I.e. you have to opt-in and build performance counters. You @lgirdwood originally asked for these, so please ack/nack the change.

Reply:
ack

Reply:
Btw, we do need to keep the over budget warnings/errors in the trace for user bug reports.

Reply:
Ack @lgirdwood . We still keep the counter and checks in zephyr_domain_thread_fn(). This emit data on overload load on each DSP core, and emits a warning if the budget is overruns (the "overruns:xx" counter). This has proven useful to debug system issue (e.g. some driver mishaving, unexpected IPC sequences, etc system level issues).

Reply:
ok, will only keep build in for core, for task and component, it will depend on user enable -o app/perf_overlay.conf during build, currently are focusing on the comp_copy from header file to c source file, cmoka have some errors, in fixing.